templates/glue-workflow.j2

```diff
@@ -89,6 +103,29 @@ Resources:
       StartOnCreation: true
       WorkflowName: !Ref PrimaryWorkflow
 
+  JsontoParquetCompleteTrigger:
```

Comment:
We want this resource as well as the triggered glue job to be conditional on whether the namespace is `staging`.

Reply:
The latter, and thanks for pointing this out. I almost forgot about this. :')

Reply:
@philerooski 

From an infrastructure perspective, because I made our compare-parquet glue job into a separate resource, it will get created for any namespace no matter what just not triggered when I add in the condition to this trigger, however is it alright if the resource gets created when deploying to main. It would make it confusing when we look at the glue workflow console for a `main` workflow and see a non-active job...

Otherwise, I would create something like two conditions in the generic `glue-job-python-shell.yaml` that's like `IsNamespaceDependent` and `IsStagingNamespace` so we can only create the compare parquet dataset on specific conditions. But yeah this would involve updating the s3 to json stack configs to fullfil that. 

Or a 3rd way that I haven't considered? Maybe adding the compare parquet to the glue workflow template but that means a ton of parameters and code gets added there...

Do we have a preferred method?

Reply:
Agree with your Edit @rxu17

Reply:
**Update:** After some fiddling around, the 1st way seems the least complex mainly because if we conditionally create the compare parquet glue job, because it's a separate resource, there's just a lot of dependency errors such as the compare parquet job name that is being used by the glue workflow stack that it can't bypass.

On the bright side, with the 1st way, what happens is the compare parquet job does get created but it doesn't show up in the glue workflow console at all when namespace is main because it's not being triggered by anything and will only run on a "feature" namespace. 

That is the resolution i have found for now.

Reply:
We use the `!Ref AWS::NoValue` method in `templates/s3-bucket.yaml`

Reply:
Right, I see it under `BucketName` under `Properties` for the Bucket resource. I think it still creates the Bucket resource, just doesn't output a BucketName.